components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -5318,6 +5346,39 @@ private int getAppIdUsingResourceId(String resourceId, String tenantDomain)
         return applicationId;
     }
 
+    /**
+     * Returns the internal application id for a given resourceId in a tenant.
+     *
+     * @param resourceId
+     * @return
+     * @throws IdentityApplicationManagementException
+     */
+    private int getAppIdUsingResourceId(String resourceId)
+            throws IdentityApplicationManagementException {
+
+        int applicationId = -1;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+
+            try (NamedPreparedStatement statement = new NamedPreparedStatement(connection,
+                    ApplicationMgtDBQueries.LOAD_APP_ID_BY_UUID)) {
```

Comment:
You can use one try block with multiple resources 
try (Connection connection = IdentityDatabaseUtil.getDBConnection(false);
NamedPreparedStatement statement = new NamedPreparedStatement(connection,
                    ApplicationMgtDBQueries.LOAD_APP_ID_BY_UUID))